src/plugins/validation/oas3/semantic-validators/security-definitions-ibm.js

```diff
@@ -0,0 +1,192 @@
+          path,
+          authId: key
+        });
+      }
+      // oauth2 validation
+      else if (type === OAUTH2) {
+        const flows = security.flows;
+        const tokenUrl = security.tokenUrl;
+
+        if (!flows) {
+          errors.push({
+            message:
+              "oauth2 authorization must have required 'flows'. Valid values are 'implicit', 'password', 'clientCredentials' or 'authorizationCode'",
+            path,
+            authId: key
+          });
+        } else if (
+          !flows.implicit &&
+          !flows.authorizationCode &&
+          !flows.password &&
+          !flows.clientCredentials
+        ) {
+          errors.push({
+            message:
+              "oauth2 authorization must have required 'flows'. Valid values are 'implicit', 'password', 'clientCredentials' or 'authorizationCode'",
+            path,
+            authId: key
+          });
+        } else if (flows.implicit) {
+          const authorizationUrl = flows.implicit.authorizationUrl;
+          if (!authorizationUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'authorizationUrl' parameter if type is `implicit` or `authorizationCode`.",
+              path,
+              authId: key
+            });
+          } else if (!flows.implicit.scopes) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'scopes' property.",
+              path,
+              authId: key
+            });
+          }
+        } else if (flows.authorizationCode) {
+          const authorizationUrl = flows.authorizationUrl;
+          if (!authorizationUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization implicit flow must have required 'authorizationUrl' parameter if type is `implicit` or `authorizationCode`.",
+              authId: key
+            });
+          }
+        } else if (flows && flows.password) {
+          const tokenURL = flows.tokenURL;
+          if (!tokenURL) {
+            errors.push({
+              message:
+                "oauth2 authorization password flow must have required 'tokenUrl' string parameter.",
+              path,
+              authId: key
+            });
+          }
+        } else if (flows === CLIENT_CREDENTIALS) {
+          if (!tokenUrl) {
+            errors.push({
+              message:
+                "oauth2 authorization clientCredentials flow must have required 'tokenUrl' string parameter.",
+              path,
+              authId: key
+            });
+          }
+        }
+      } else if (type === HTTP) {
+        //scheme is required
+        if (!security.scheme) {
+          errors.push({
+            message:
+              'scheme must be defined for type `http` and must be a string',
```

Comment:
I don't think it's necessary to say "must be a string" but I know the spec requires a string so if you really want to enforce it, make sure you check for it. This applies to the other checks too